src/containers/DataDisplay/DataViewTable.tsx

```diff
@@ -1,86 +1,176 @@
     dataViewCode,
     apiQueries,
+    filter: gqlFilter,
   })
-  usePageTitle(dataViewTable?.title || '')
+  const title = location?.state?.title ?? dataViewTable?.title ?? ''
+  usePageTitle(title)
 
   useEffect(() => {
     setApiQueries(getAPIQueryParams(query))
+    setGqlFilter(buildQueryFilters(query, filterDefinitions))
   }, [query])
 
+  useEffect(() => {
+    updateQuery({ search: debounceOutput })
+  }, [debounceOutput])
+
   if (error) {
     return <Message error header={strings.ERROR_GENERIC} content={error.message} />
   }
 
-  if (loading || !dataViewTable) return <Loading />
-
-  const showDetailsForRow = (id: number) => push(`/data/${dataViewCode}/${id}`)
-
-  const { headerRow, tableRows, title, totalCount } = dataViewTable
-
   return (
     <div id="data-view">
-      <Header as="h4">{title}</Header>
       <div id="list-container" className="data-view-table-container">
-        <Table stackable selectable>
-          <Table.Header>
-            <Table.Row>
-              {headerRow.map(({ title }: any) => (
-                <Table.HeaderCell key={title} colSpan={1}>
-                  {title}
-                </Table.HeaderCell>
-              ))}
-            </Table.Row>
-          </Table.Header>
-          <Table.Body>
-            {tableRows.map(({ id, rowValues }: { id: number; rowValues: any }) => (
-              <Table.Row
-                key={`row_${id}`}
-                className="clickable"
-                onClick={() => showDetailsForRow(id)}
-              >
-                {rowValues.map((value: any, index: number) => (
-                  <Table.Cell key={`value_${index}`}>
-                    {getCellComponent(value, headerRow[index], id)}
-                  </Table.Cell>
-                ))}
-              </Table.Row>
-            ))}
-          </Table.Body>
-        </Table>
-        <PaginationBar
-          totalCount={totalCount}
-          perPageText={strings.OUTCOMES_TABLE_PAGINATION_TEXT}
-          strings={strings}
-        />
+        <div className="flex-row-space-between-center" style={{ width: '100%' }}>
+          <Header as="h3">{title}</Header>
+          {dataViewTable && dataViewTable?.searchFields?.length > 0 && (
+            <Search
+              placeholder={strings.DATA_VIEW_SEARCH_PLACEHOLDER}
+              onSearchChange={(e: any) => {
+                setSearchText(e.target.value)
+                setDebounceInput(e.target.value)
+              }}
+              input={{ icon: 'search', iconPosition: 'left' }}
+              open={false}
+              value={searchText}
+            />
+          )}
+        </div>
+        <div className="flex-row-space-between-center" style={{ width: '100%' }}>
+          <ListFilters filterDefinitions={filterDefinitions} filterListParameters={{}} />
+        </div>
+        {loading && <Loading />}
+        {dataViewTable && (
+          <DataViewTableContent dataViewTable={dataViewTable} apiQueries={apiQueries} />
```

Comment:
Moved the table content into its own component, so we can render the container (and search/filters) without waiting for the table data to load.